apply.sh

```diff
@@ -49,20 +55,29 @@ fi
 cd /nix/home/darwin-config
 nix-shell -p git --run "git fetch $CONFIG_REPO $CONFIG_BRANCH && git checkout FETCH_HEAD"
 
+nix --extra-experimental-features 'nix-command flakes' profile install nixpkgs#git
+
+config="/nix/home/darwin-config"
+nixpkgs=$(nix --extra-experimental-features 'nix-command flakes' eval --raw "$config"#inputs.nixpkgs)
+darwin=$(nix --extra-experimental-features 'nix-command flakes' eval --raw "$config"#inputs.darwin)
+export NIX_PATH=darwin-config="$config/$CONFIG_TARGET":nixpkgs=$nixpkgs:darwin=$darwin
+
 if [ ! -e /etc/static/bashrc ]; then
-yes |    $(nix-build https://github.com/LnL7/nix-darwin/archive/master.tar.gz -A installer --no-out-link)/bin/darwin-installer 2>&1 | tail -n10
+    nix --extra-experimental-features 'nix-command flakes' build "$config"#darwinConfigurations."$(uname -m)".system --out-link "$config/result"
+    "$config/result/sw/bin/darwin-rebuild" switch --flake "$config"#"$(uname -m)"
 fi
 
+nix --extra-experimental-features 'nix-command flakes' profile remove 0
+
 if ! hash darwin-rebuild; then
     set +eux
     . /etc/static/bashrc
     set -eux
 fi
 
-export NIX_PATH=darwin-config="/nix/home/darwin-config/$CONFIG_TARGET":nixpkgs=channel:nixpkgs-unstable:darwin=https://github.com/LnL7/nix-darwin/archive/master.tar.gz
-
 sudo rm /etc/nix/nix.conf || true
 
-darwin-rebuild switch
+export NIX_PATH=darwin-config="$config/$CONFIG_TARGET":nixpkgs=$nixpkgs:darwin=$darwin
+darwin-rebuild switch --flake "$config"#"$(uname -m)"
```

Comment:
I wonder if we need to do the `switch` both here and above? Can we ~only do it here? We should probably have the `uname -m` bit be passed in as a CONFIG_... var for other use cases, too.

Reply:
We don't, I'm just mimicking what we had before -- the nix-darwin installer implicitly switches to a default config IIRC. But I guess that's just for bootstrapping the environment to be able to use nix-darwin properly, so I don't mind.